apps/web/src/router/index.ts

```diff
@@ -68,6 +68,26 @@ export class SpaceRouter {
             const userNeedPwdReset = SpaceRouter.router.app?.$store.getters['user/isUserNeedPasswordReset'];
             let nextLocation;
 
+
+            // Grant API Key
+            const accessToken = SpaceConnector.getAccessToken();
+            const isDuplicatedRoute = to.name === from.name;
+            if (accessToken && isTokenAlive && !isDuplicatedRoute) {
+                let scope: string;
+                if (to.name?.startsWith('admin.')) {
+                    scope = 'DOMAIN';
+                } else if (to.params.workspaceId) {
+                    scope = 'WORKSPACE';
+                } else scope = 'USER';
+                const grantRequest = {
+                    scope,
+                    workspace_id: to.params.workspaceId,
```

Comment:
Question: Are there any cases where params do not exist at all? If you have any, you may need a defense code.

Reply:
Empty `params` alway exists. And I handle no-workspaceId case above scope assignment!
In no-workspaceId case, scope is `USER`!